### PR TITLE
Add FuelRegistry

### LINKED ISSUE #37: Fuel API
Remember, forge uses an ItemStack-aware method in IForgeItem, but in fabric you just add to the map using the api.

### PR DESCRIPTION
**Pull request by shedaniel**
Close #37 
Signed-off-by: shedaniel <daniel@shedaniel.me>

### CODE REVIEW

File: forge/src/main/java/me/shedaniel/architectury/registry/fuel/forge/FuelRegistryImpl.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+import net.minecraft.world.level.ItemLike;
+import net.minecraftforge.event.furnace.FurnaceFuelBurnTimeEvent;
+import net.minecraftforge.eventbus.api.SubscribeEvent;
+
+public class FuelRegistryImpl {
+    private static final Object2IntMap<ItemLike> ITEMS = new Object2IntLinkedOpenHashMap<>();
+    
+    static {
+        EventBuses.onRegistered(ArchitecturyForge.MOD_ID, bus -> {
+            bus.register(FuelRegistryImpl.class);
```
Comment (MaxNeedsSnacks) [inline forge/src/main/java/me/shedaniel/architectury/registry/fuel/forge/FuelRegistryImpl.java]: `FurnaceFuelBurnTimeEvent` is fired on the `MinecraftForge.EVENT_BUS`, not the mod bus, see https://github.com/MinecraftForge/MinecraftForge/blob/b6ce4fb066eb886a305f0678649c33d6f8ba4f76/src/main/java/net/minecraftforge/event/furnace/FurnaceFuelBurnTimeEvent.java#L42

**Review by MaxNeedsSnacks — CHANGES_REQUESTED** [review]

**Review by MaxNeedsSnacks — APPROVED** [review]

**Review by Trikzon — APPROVED** [review]
Looks good to me and is pretty close to what I have done myself. Only real difference is passing an `ItemStack` instead of an `Item` to `get()`. I'm not sure why a stack would hold a different burn time than its item, but it's nice to see it supported.

**Comment by MaxNeedsSnacks** [comment]
That is to futureproof our api for stack-aware approaches down the line, should FAPI add support for them eventually

A situation where a stack may have a different burn time than the base item would be something like Extra Utilities‘ Redstone Coil, which would extract from the player‘s power network to heat a furnace

**Comment by shedaniel** [comment]
Plus forge needs it anyways.

**Pull request merged**